modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowDefinitionManagerTest.java

```diff
@@ -74,6 +74,65 @@ public void testDeployGroovyWorkflowDefinition() throws Exception {
 		}
 	}
 
+	@Test
+	public void testDeployJavaWorkflowDefinition() throws Exception {
+
+		// Custom workflow definition, Java based scripts,
+		// allowScriptContentToBeExecutedOrIncluded false
+
+		String content1 = StringUtil.read(
+			getResourceInputStream(
+				"java-based-condition-node-workflow-definition.xml"));
+
+		try (Closeable closeable =
+				ScriptManagementConfigurationTestUtil.disable()) {
+
+			AssertUtils.assertFailure(
+				KaleoDefinitionValidationException.NotAllowedScriptLanguage.
+					class,
+				"Java is not allowed",
+				() -> _workflowDefinitionManager.deployWorkflowDefinition(
+					TestPropsValues.getCompanyId(), TestPropsValues.getUserId(),
+					StringPool.BLANK, "Java Workflow Definition",
+					content1.getBytes()));
+		}
+
+		// Custom workflow definition, Java based scripts,
+		// allowScriptContentToBeExecutedOrIncluded true
+
+		Assert.assertNotNull(
+			_workflowDefinitionManager.deployWorkflowDefinition(
+				TestPropsValues.getCompanyId(), TestPropsValues.getUserId(),
+				StringPool.BLANK, "Java Workflow Definition",
+				content1.getBytes()));
+
+		// System workflow definition, Java based scripts,
+		// allowScriptContentToBeExecutedOrIncluded false
+
+		String content2 = StringUtil.read(
+			getResourceInputStream(
+				"message-boards-moderation-workflow-definition.xml"));
```

Comment:
@brianchandotcom `message-boards-user-stats-moderation` is a system workflow definition, so we assume that the java based script is safe to be execute since it is provided but us.